// ngx-ionic-image-viewer/esm5/lib/ngx-ionic-image-viewer.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngx-ionic-image-viewer.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { CommonModule } from '@angular/common';
import { IonicModule } from '@ionic/angular';
import { NgModule } from '@angular/core';
import { NgxIonicImageViewerComponent } from './ngx-ionic-image-viewer.component';
import { NgxIonicImageViewerDirective } from './ngx-ionic-image-viewer.directive';
import { ViewerModalComponent } from './viewer-modal/viewer-modal.component';
import * as ɵngcc0 from '@angular/core';
var NgxIonicImageViewerModule = /** @class */ (function () {
    function NgxIonicImageViewerModule() {
    }
NgxIonicImageViewerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxIonicImageViewerModule });
NgxIonicImageViewerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxIonicImageViewerModule_Factory(t) { return new (t || NgxIonicImageViewerModule)(); }, imports: [[CommonModule, IonicModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxIonicImageViewerModule, { declarations: function () { return [NgxIonicImageViewerComponent, NgxIonicImageViewerDirective, ViewerModalComponent]; }, imports: function () { return [CommonModule, IonicModule]; }, exports: function () { return [NgxIonicImageViewerComponent, NgxIonicImageViewerDirective, ViewerModalComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxIonicImageViewerModule, [{
        type: NgModule,
        args: [{
                declarations: [NgxIonicImageViewerComponent, NgxIonicImageViewerDirective, ViewerModalComponent],
                imports: [CommonModule, IonicModule],
                entryComponents: [ViewerModalComponent],
                exports: [NgxIonicImageViewerComponent, NgxIonicImageViewerDirective, ViewerModalComponent]
            }]
    }], function () { return []; }, null); })();
    return NgxIonicImageViewerModule;
}());
export { NgxIonicImageViewerModule };
