// ionic-long-press/esm5/long-press.module.js
import * as tslib_1 from "tslib";
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LongPressDirective } from './directives/long-press.directive';
import * as ɵngcc0 from '@angular/core';
var LongPressModule = /** @class */ (function () {
    function LongPressModule() {
    }
LongPressModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: LongPressModule });
LongPressModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function LongPressModule_Factory(t) { return new (t || LongPressModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(LongPressModule, { declarations: function () { return [LongPressDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [LongPressDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LongPressModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [LongPressDirective],
                exports: [LongPressDirective]
            }]
    }], function () { return []; }, null); })();
    return LongPressModule;
}());
export { LongPressModule };
